dips/dip-10.md

```diff
@@ -101,130 +110,87 @@ struct DiemIdDomainEvent {
-
-* Pay endpoint is used to exchange details of the payment such as information about Sender and Receiver off the chain, and negotiate `Payment_Reference_ID`
-
-* Once the sender learns Payment_Reference_ID, it will submit an on-chain transaction including the `Payment_Reference_ID` as a metadata. Receiver will use the `Payment_Reference_ID` in the transaction to link the on-chain transaction to the payment.
+**CommandResponseObject:**
 
-* Sender will submit a transaction to the `receiver_address` on-chain account in the response. This account address can be either Parent VASP, or any of Child VASP accounts of the receiver VASP. This is needed to allow the receiver VASP to manage load between different on-chain accounts.
+| Field 	    | Type 	     | Required? 	| Description 	           |
+|-------	    |------	     |-----------	|-------------	           |
+| _ObjectType   | str        | Y | Fixed value: `CommandResponseObject`|
+| status       | str | Y | Either `success` or `failure`. |
+| result       | Result object | Y | The Result obejct of response. |
+| cid           | str         | Y            | A unique identifier for the Command. Should be a UUID according to [RFC4122](https://tools.ietf.org/html/rfc4122) with "-"'s included. |
 
-* `Payment_Reference_ID` must be unique for each payment, it must not be reused in any way to avoid leaking privacy information on-chain.
+**CommandResultObject:**
 
-* `Payment_Reference_ID` returned by this endpoint is not signed for KYC purposes. This means that payment above KYC threshold can not be submitted right away using the provided Payment_Reference_ID. Instead, sender VASP needs to proceed to KYC endpoint to negotiate the signed Reference ID.
+| Field 	    | Type 	     | Required? 	| Description 	           |
+|-------	    |------	     |-----------	|-------------	           |
+| _ObjectType   | str        | Y | Fixed value: `ReferenceIDCommandResponse`|
+| recipient_address       | str | Y | Receiver's on-chain address |
 
-* KYC endpoint API will be updated to accept `Payment_Reference_ID` returned by the Diem ID endpoint instead of address/subaddress.
-
-## User Object
-User Object defines reach information about the user.
-
-All fields, except for `diem_id`, in this object are optional - VASP can choose to share either of those fields or none.
-
-Additionally, all fields in this object are informational - **they must not be used for KYC and only intended for UI purposes**.
-```json
-{
-   "diem_id": "<Diem_ID>",
-   "display_name": "<Display name>",
-   "profile_picture": "<profile_picture_uri>"
+If the amount is below the travel rule limit, the sending VASP can send a p2p transaction with PaymentMetadata and the `reference_id` VASPs agreed on to settle the transaction. 
+```
+enum PaymentMetadata {
+    PaymentMetadataV0(ReferenceId),
 }
+type ReferenceId = [u8, 16];
```

Comment:
do we want to remove this line?

Reply:
No this defines the onchain structure of the reference id, which is a uuid in a compact form.

Reply:
why?